File: src/main/java/dev/anvilcraft/addon/template/AddonConfig.java
package dev.anvilcraft.addon.template;

import dev.anvilcraft.lib.v2.config.BoundedDiscrete;
import dev.anvilcraft.lib.v2.config.Comment;
import dev.anvilcraft.lib.v2.config.Config;

@Config(name = AnvilCraftAddonTemplate.MOD_ID)
public class AddonConfig {
    @Comment("Whether to log the dirt block on common setup")
    public boolean logDirtBlock = false;

    @Comment("A magic number")
    @BoundedDiscrete(max = 24, min = 2)
    public int magicNumber = 2;

    @Comment("What you want the introduction message to be for the magic number")
    public String magicNumberIntroduction = "";
}

File: src/main/kotlin/dev/anvilcraft/addon/template/AddonConfig.kt
package dev.anvilcraft.addon.template

import dev.anvilcraft.lib.v2.config.BoundedDiscrete
import dev.anvilcraft.lib.v2.config.Comment
import dev.anvilcraft.lib.v2.config.Config

@Config(name = AnvilCraftAddonTemplate.MOD_ID)
class AddonConfig {
    @Comment("Whether to log the dirt block on common setup")
    var logDirtBlock: Boolean = false

    @Comment("A magic number")
    @BoundedDiscrete(max = 24.0, min = 2.0)
    var magicNumber: Int = 2

    @Comment("What you want the introduction message to be for the magic number")
    var magicNumberIntroduction: String = ""
}